File: examples/modulith-clinical-tool-jpa/src/test/java/io/zenwave360/example/clinicaltool/modules/users/config/RepositoriesInMemoryConfig.java
package io.zenwave360.example.clinicaltool.modules.users.config;

import io.zenwave360.example.clinicaltool.modules.users.*;
import io.zenwave360.example.clinicaltool.modules.users.inmemory.*;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Primary;

// @Configuration
public class RepositoriesInMemoryConfig {

    protected final UserRepository userRepository = new UserRepositoryInMemory();

    @Bean
    @Primary
    public <T extends UserRepository> T userRepository() {
        return (T) userRepository;
    }
}

File: examples/modulith-klinical-tool-jpa/src/test/kotlin/io/zenwave360/example/clinicaltool/modules/users/config/RepositoriesInMemoryConfig.kt
package io.zenwave360.example.clinicaltool.modules.users.config

import io.zenwave360.example.clinicaltool.modules.users.*
import io.zenwave360.example.clinicaltool.modules.users.inmemory.*
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Primary

// @Configuration
open class RepositoriesInMemoryConfig {

    protected val userRepository = UserRepositoryInMemory()

    @Bean
    @Primary
    fun userRepository(): UserRepositoryInMemory {
        return userRepository
    }
}
